prow/cmd/deck/job_history.go

```diff
@@ -160,39 +172,42 @@ func readJSON(bucket storageBucket, key string, data interface{}) error {
 }
 
 // Lists the GCS "directory paths" immediately under prefix.
-func (bucket gcsBucket) listSubDirs(prefix string) ([]string, error) {
+func (bucket gcsBucket) listSubDirs(ctx context.Context, prefix string) ([]string, error) {
 	if !strings.HasSuffix(prefix, "/") {
 		prefix += "/"
 	}
+	it, err := bucket.Opener.Iterator(ctx, fmt.Sprintf("%s://%s/%s", bucket.storageProvider, bucket.name, prefix), "/")
+	if err != nil {
+		return nil, err
+	}
+
 	dirs := []string{}
-	it := bucket.Objects(context.Background(), &storage.Query{
-		Prefix:    prefix,
-		Delimiter: "/",
-	})
 	for {
-		attrs, err := it.Next()
-		if err == iterator.Done {
+		done, attrs, err := it.Next(ctx)
+		if done {
```

Comment:
Yup, you're of course right about the error checking. I would be fine with returning an error instead of the additional bool var. The only question is which one. The internally used `*storage.ObjectIterator` returns `iterator.Done`; `*blob.ListIterator` returns `io.EOF`. I would suggest defining a new error in `iterator.go` in the same style as `iterator.Done`:
```
var IteratorDone = errors.New("no more items in iterator")
```

Reply:
I am fine with both `io.EOF` or a custom error type

Reply:
Fixed